### ISSUE TITLE
One-to-many table interface isn't showing newly created entries' many-to-one values

### ISSUE THREAD

**Issue by jagyas** [directus/directus#19295]
### Describe the Bug

I am having a many to one relation and when selecting values it is not visible in table view. Once the item is saved then it is visible. Is this a normal behaviour or it is a bug?

![image](https://github.com/directus/directus/assets/6293136/e81a0410-c2f4-499b-913e-79ca9e66bba7)

### To Reproduce

1. Create a collection (child) and create a many to one field inside it.
2. Refer this collection in any other collection (parent) with a one to many field and change interface of this field to table view.
3. Once done try to add child collection in parent collection (See attached screenshot in description where "Variations" collection is child). It will show blank first but once values are saved then it will be visible. 

### Directus Version

v10.5.2

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by DMachad0** [comment]
same issue

**Comment by DanielBiegler** [comment]
Does this happen for you when **creating** a child from the parent - and or when **selecting** a child from parent?

The former for me looks like you described, the latter works for me.

**Comment by DanielBiegler** [comment]
Inspecting the raw value reveals that we see the childs related foreign key which we could technically use to resolve the needed fields. Yeah, I think this would be nice to have.

**Comment by jagyas** [comment]
Yeah. It feels important to me because I need to select multiple values (can go up to 10 or more items) and I will be not having any clue what all values I have already selected or I am selecting duplicate one by mistake.

**Comment by Nitwel** [comment]
Note to self: We would have to implement a new function that tires to load relational fields on newly created items in a m2m/o2m/m2a. Might be a really hacky solution, don't know yet.

**Comment by vonadz** [comment]
There's a similar issue for create when the child item has a many to one relationship to another collection.

For example:

Collection "product":
id
specification_ids (one to many with specifications, interface is table with specification_names.name and specifications.value)

Collection "specifications":
id
specification_name_id (many to one with specification_names)
value

Collection "specification_names":
id
name

Creating a new specification_ids item on product shows the specification_names.name value is blank in the table interface until saved.

**Comment by guycalledseven** [comment]
Any progress maybe? This is really bothering our users.

It happens with any kind of relation actually. 

What I figured out is that if you have collection that has relation on some field, and you are editing main collection and:

1. If you add existing item in related field in main collection the bug happens (related fields are not visible in newly added block on page just before save is pressed). when you press save on main collection, relation appears correctly. 

![bug when adding existing items](https://i.imgur.com/46gbLdb_d.webp?maxwidth=760&fidelity=grand)

2. if you add NEW item in related field in main collection, this bug **does not** appear. The related fields from related collection are shown correctly before the save.

![no bug when adding new item to related field](https://i.imgur.com/Zut6HQn_d.webp?maxwidth=760&fidelity=grand)

After save of main collection, the related field is shown as defined in interface settings of that field:
![after save](https://i.imgur.com/j9aLfw2_d.webp?maxwidth=760&fidelity=grand)

On main collection, I have O2M field to other collection. That other collection has OTHER collection connected to O2M as well.  Only 1 field is shown from related collection.

**Comment by xaviertp** [comment]
Hi , any updates or planning on this? This is quite a common case to use the CMS an it really brother the users a lot. 
The users won't know what they are creating unless they click into the item one by one which make the listing become useless.

**Comment by yipivan** [comment]
I can confirm this is a major issue we encounter during development. This causes our users cannot preview the content they created in the M2O collection from the display templates. This is a turndown for us, any fix or work around would be nice.

**Comment by xaviertp** [comment]
@rijkvanzanten ,  still not get any feedback or schedule on it since a year ago and it should bother users for year too. thank you for the concern.

**Comment by rijkvanzanten** [comment]
@xaviertp A PR is more than welcome 👍🏻

**Comment by pasqualepalmaccio** [comment]
I join the chorus and share the concerns about this issue. It's unacceptable that this component functions in this way; the data needed to display the information in the template is already available in the context and shouldn't necessarily be persisted in the collection related to the main one. I can't find alternative solutions; this is solely a problem with the Directus app, and it can't be magically fixed with other solutions like custom extensions, behind-the-scenes updates, etc.

**Comment by pasqualepalmaccio** [comment]
@rijkvanzanten @jagyas 

I have found a solution. In summary

Problem Description: Currently, Directus has a limitation when handling one-to-many (O2M) relationships between a primary collection and a pivot collection that, in turn, expresses a relationship (many-to-one or others) with another collection. During the creation of a record in the primary collection, the raw value of the field in the pivot collection associated with the M2O relationship is not fully populated and contains only the reference key (ID) of the related item.

This limitation prevents users from viewing the complete data of the related item until the primary collection is saved, and the data is actually written to the database. Only after saving can the complete properties of the associated item be accessed through the pivot collection.

Proposed Solution: I have found that this limitation can be overcome by expanding the raw value during the record creation phase, allowing the display of all required fields of the final associated item before saving. This enhancement would allow for a smoother and more complete handling of relational data, providing an accurate preview of the data without the need to save the record first.

Example Implementation: I developed a solution by modifying the interface described in the official Directus documentation [at this link](https://docs.directus.io/guides/extensions/interfaces-relational-dropdown.html). The primary modification was made in the setDropdown function, where I added an extra API call to retrieve the complete data of the selected item, thus populating the raw value with all necessary fields.

async function setDropdown(item) {
  if (item == null) {
    searchQuery.value = item;
    emit('input', item);
  } else {
    // Perform an API call to get the full data of the selected item
    const selectedItem = await api.get(`/items/${relatedCollection.value.collection}/${item[primaryKey.field]}`);
    
    // Populate the raw value with the additional fields
    const completeItem = selectedItem.data.data;
    searchQuery.value = outputFields(completeItem);
    emit('input', {
      ...completeItem,
      [primaryKey.field]: item[primaryKey.field],
    });
  }

  fetchResults();
}

Conclusion: I request the implementation of a similar default function to handle relational fields in pivot collections during record creation, improving the user experience and relational data management in Directus. My solution shows that it is possible to display complete data already at the creation stage, and I believe that a default implementation could benefit all users working with complex relationships in Directus.

**Comment by sksar** [comment]
Any solution to this yet ? Anything temporary even ?

**Comment by formfcw** [comment]
We haven't been able to reproduce this based on the given information, so I'll close this for now. This issue has probably been fixed in the meantime. Happy to keep discussing in this thread, and we'll reopen if this issue still persists.

**Comment by Novex** [comment]
@formfcw Unfortunately it has not been fixed - it's easy to reproduce this on https://directus.pizza

1. Edit a relationship on a collection to reference a nested field, either in the title or as a table column eg:
   * Go to https://sandbox.directus.io/admin/settings/data-model/help_collections/articles -> Interface -> Display Template -> Edit Raw Value and change `{{title}}` to `{{title}} by {{owner.email}}` and save the field definition
2. Load the collection, create a new item in the relationship and fill out the nested field, eg:
   * Go to https://sandbox.directus.io/admin/content/help_collections/+ -> Scroll down to `Articles` -> Create New -> Fill out the Article Title -> Add an existing Owner with an email address (eg. CRM Demo User), Save the user selection, Save the Article record
3. Notice that the Articles list is missing the user's email, replaced by `--`, eg. `Some Article Title by --`
3. Fill out the Title and Save and stay on the screen, notice that the Articles list now reads correctly, eg. `Some Article Title by admin+crm@example.com`

Expected behaviour as mentioned earlier is that the nested field should display the value too just like a non-nested field.

I have created a video that shows the problem as well if that is clearer:

![Image](https://github.com/user-attachments/assets/823e79b4-e5f4-4c7b-938e-2760bf2bfc72)

**Comment by formfcw** [comment]
The original issue here cannot be reproduced. However, there is still an issue with displaying the values of related items. A duplicate issue more accurately describes the problem. That's why I am closing this one in favor of the other.
